railties/lib/rails/generators/app_base.rb

```diff
@@ -404,6 +412,49 @@ def run_bundle
         bundle_command("install") if bundle_install?
       end
 
+      def npm_path
+        commands = ["npm"]
+
+        if RbConfig::CONFIG["host_os"] =~ /mswin|cygwin/
+          ENV['PATHEXT'].split(File::PATH_SEPARATOR).each do |ext|
+            commands << commands[0] + ext
+          end
+        end
+
+        npm_path = commands.find do |cmd|
+          paths = ENV['PATH'].split(File::PATH_SEPARATOR)
+          path = paths.find do |p|
+            full_path = File.expand_path(cmd, p)
+            File.executable?(full_path) && File.file?(full_path)
+          end
+          path && File.expand_path(cmd, path)
+        end
+
+        unless npm_path
+          puts %{Warning: npm option passed but npm executable was 
+            not detected in the system. Download Node.js in https://nodejs.org/en/download/}
```

Comment:
Let's move this warning message to https://github.com/rails/rails/pull/26836/files#diff-248322211d56bb3b2ab59ac68a2839f5R212, right now it's being displayed everytime the npm_path function is called if command is not found